src/Resources/config/actions.xml

```diff
@@ -1,15 +1,14 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <!-- NEXT_MAJOR: remove the "class" and "public" attributes -->
-        <service id="Sonata\AdminBundle\Action\DashboardAction" class="Sonata\AdminBundle\Action\DashboardAction" public="true">
+        <service id="Sonata\AdminBundle\Action\DashboardAction" public="true">
```

Comment:
```suggestion
        <service id="Sonata\AdminBundle\Action\DashboardAction">
```
Based on the comment

Reply:
I did a research on this topic, about this NEXT_MAJOR comment:

@greg0ire added it in the first place, when there were only 2 classes here:
https://github.com/sonata-project/SonataAdminBundle/pull/5120

Then it seems like it was not a good idea to use the FQCN as the id, so that is why there are services here with normal id and FQCN id:
https://github.com/sonata-project/SonataAdminBundle/pull/5153#discussion_r204208506

Another reference of this topic:
https://github.com/symfony/symfony-docs/pull/9886#issuecomment-399927872

So... We should not use FQCN as id, because its not that good on reusable bundles (you should use in your own app, along with autoconfigure and autowiring)

I will do a change here to add again the class, remove the public = true (I thought you need controller to be public in order to retrieve them with the routing, will test it), and change the routing to use the new Identifiers.

Reply:
Another discovery on my side... You don't have to make them public IF you add the tag: "controller.service_arguments"

That is nice! So I added it to on every action in this file.

Reply:
Another PR to read if you want to know more about service id vs class: https://github.com/symfony/symfony-docs/pull/10109

Reply:
That is an interesting topic @sonata-project/contributors becase we are merging some PRs which introduce services with id === class, and seems like a bad practice for reusable bundles.

Reply:
ping @wbloszyk and @phansys , because you have been fixing the rest api and in the process I think we introduced some aliases or classes with FQCN, maybe that was not a good idea

Reply:
@jordisala1991 Seems like indeed https://github.com/symfony/symfony-docs/pull/9886#issuecomment-399927872

Reply:
> Another discovery on my side... You don't have to make them public IF you add the tag: "controller.service_arguments"

We don't use services as arguments for our actions though, do we? Don't we pass every service in the constructor every time?

Reply:
I was talking about route definition. You need to make them public (old solution applied to 3.x) or use the controller.service_arguments tag.

Reply:
Yes, but using that tag does not really make sense if no services are passed to methods, does it?

Reply:
Well it does not, but it removes the error I got when not using public true

Reply:
https://github.com/symfony/symfony/pull/24126

Here is the behavior. 

We either discard the next major comment about remove public or we follow that comment and we need to add the tag, not sure which one is better, to me having public true is, because its explicit.

Wdyt?

Reply:
To me if we are just using the tag to make them public, I would set explicitly to `public` too and remove the tag.

Reply:
Done